plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/BOAndDPCompileHook.java

```diff
@@ -52,6 +46,22 @@ public void validate(Map<String, Object> toCheck) throws Exception {
 
             File pluginDir = (File) moreInfo.get("pluginDir");
 
+            Path pluginSourcesDir = config.getLocalCheckoutDir().toPath();
+
+            if (pluginSourcesDir != null) {
+                boolean isMultipleLocalPlugins = config.getIncludePlugins() != null && config.getIncludePlugins().size() > 1;
+                // We are running for local changes, let's copy the .eslintrc file if we can
+                //If we are using localCheckoutDir with multiple plufins the .eslintrc must be located at the top level
+                // If not it must be located on the parent of the localCheckoutDir
+                if (!isMultipleLocalPlugins) {
+                    pluginSourcesDir = pluginSourcesDir.getParent();
+                }
+                // Copy the file if it exists
+                Files.walk(pluginSourcesDir, 1)
```

Comment:
Can't you access to the file directly? It seems unnecessary to walk the whole folder when you already know what you are looking for.

Reply:
@varyvol Yeah, I can do an if and things like that, but I think this is much readable and that walking is not a very big performance penalty because:

* The stream is lazily populated with paths, no reading files at all
* I am only walking the first level, is not recursive
* File.exists will require a call to the underlying filesystem the same for getting a lazy list of files, so I am not even sure if there is performance penalty at all, also I am not sure if File is lazy as are the Paths returned by the walk method 

So, in this case, I prefer this small stream code that is perfectly legible than an if else structure, and I am not sure even if this could be a performance penalty.